### PR TITLE
Prevent OTP 24 compile-time warning

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
(parse_trans required bumping)

- [x] wait until erlef/setup-beam is available with `rebar3` then pull it here
- [x] add 24 to CI, after https://github.com/erlef/setup-beam/pull/34

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
Hm... GitHub action is broken. Moving to a working container-based approach.

**Comment by paulo-ferraz-oliveira** [comment]
I had to bump CI min. to OTP 21 due to `hackney` pulling a `parse_trans` version that requires it. Is this acceptable?

**Comment by tsloughter** [comment]
What about the erlef/setup-elixir action instead of the containers?

**Comment by paulo-ferraz-oliveira** [comment]
> What about the erlef/setup-elixir action instead of the containers?

😄 not yet finished for `rebar3`. But this can wait, sure.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -15,15 +15,15 @@ jobs:
 
     strategy:
       matrix:
-        otp_version: ['23.1', '22.3.4.2', '21.3.8.16', '20.3.8.21']
+        otp_version: ['23', '22', '21']
```
Comment (yurrriq) [inline .github/workflows/main.yml]: can we add back R20 now?

Comment (paulo-ferraz-oliveira) [reply .github/workflows/main.yml]: If `hackney`'s no longer forcing a min. of 21, sure.

**Edit**: but I guess it still is, via `parse_trans`: https://github.com/uwiger/parse_trans/blob/master/rebar.config#L18.

**Comment by paulo-ferraz-oliveira** [comment]
Any chance this gets merged and tagged any time soon? Thanks.

**Comment by paulo-ferraz-oliveira** [comment]
Any chance this gets merged and tagged any time soon? Thanks.

File: .github/workflows/main.yml (hunk 1 above)
Comment (paulo-ferraz-oliveira) [inline .github/workflows/main.yml]:
```suggestion
        otp_version: ['24', '23', '22', '21']
```

**Comment by paulo-ferraz-oliveira** [comment]
Now that OTP 24 is out, is it possible to re-run the CI, so I can move toward finalising this? CI (just now) was failing for inexistent Erlang version (setup-beam downloads from Hex.pm). It is possible the version was not there yet.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -15,15 +15,15 @@ jobs:
 
     strategy:
       matrix:
-        otp_version: ['23.1', '22.3.4.2', '21.3.8.16', '20.3.8.21']
+        otp_version: ['24', '23', '22', '21']
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/main.yml:18]:
```suggestion
        otp_version: ['23', '22', '21']
```

Comment (paulo-ferraz-oliveira) [inline .github/workflows/main.yml:18]:
```suggestion
        otp_version: ['24.0', '23', '22', '21']
```

File: .github/workflows/main.yml (hunk 1 above)
Comment (tsloughter) [reply .github/workflows/main.yml]: Why add 20?

**Comment by tsloughter** [comment]
Not sure how to force it to rerun all the CI. Can you maybe rebase and do a new push? Or create a new PR?

**Comment by paulo-ferraz-oliveira** [comment]
> Not sure how to force it to rerun all the CI. Can you maybe rebase and do a new push? Or create a new PR?

Yeah, my recent push recreated the CI, but it's broken elsewhere. Will have to dig in later.

**Comment by paulo-ferraz-oliveira** [comment]
Using 24.0 here eventually lead to a fix in `setup-beam`: https://github.com/erlef/setup-beam/pull/34.

Will wait for that and then push an update to CI.

**Comment by paulo-ferraz-oliveira** [comment]
Ready for re-review, now that CI's passing.

**Comment by paulo-ferraz-oliveira** [comment]
Bump.

**Comment by paulo-ferraz-oliveira** [comment]
Bump.

**Comment by paulo-ferraz-oliveira** [comment]
Bump.

**Review by tsloughter — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
Thank you, Tristan.

**Pull request merged**